src/factory/mastercopyDeployer.ts

```diff
@@ -7,51 +7,115 @@ import { getSingletonFactory } from "./singletonFactory";
  *
  * To get the same address on any chain.
  * @param hre hardhat runtime environment
- * @param mastercopyContractFactory
- * @param args
- * @returns The address of the deployed module mastercopy
+ * @param mastercopyContractFactory mastercopy to deploy
+ * @param args the arguments to pass to the mastercopy's constructor
+ * @returns The address of the deployed module mastercopy or undefined if it was already deployed
  */
 export const deployMastercopy = async (
   hre: HardhatRuntimeEnvironment,
   mastercopyContractFactory: ContractFactory,
   args: Array<any>,
   salt: string
-) => {
+): Promise<string | undefined> => {
   const deploymentTx = mastercopyContractFactory.getDeployTransaction(...args);
+
   if (deploymentTx.data) {
-    await deployMastercopyWithInitData(hre, deploymentTx.data, salt);
+    return await deployMastercopyWithInitData(hre, deploymentTx.data, salt);
+  }
+  throw new Error("No deployment data found");
+};
```

Comment:
Minor: I wouldn't use the word "found" here. It's more like we couldn't compute/create the deployment data